### ISSUE TITLE
cell-var-from-loop gives some false positives

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#233]
Originally reported by: **Claudiu Popa (BitBucket: [PCManticore](http://bitbucket.org/PCManticore), GitHub: @PCManticore)**

---

When defining and calling a function inside a for loop. 
For instance, here's an example:

```
#!python

for i in range(10):
   print((lambda x: i + x)(1))
```

Also, the same happens if you use the variable inside a generator and the
generator is evaluated in the body of the loop, like in the following:

```
#!python

for i in range(10):
   do_something(*map(lambda x: do something with i, an_iterable))

```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/233
